index.js

```diff
@@ -22,11 +31,13 @@ function fastifyRequestContext(fastify, opts, next) {
         ? opts.defaultStoreValues()
         : opts.defaultStoreValues
 
-    als.runWith(() => {
+    const store = defaultStoreValues ? new Map(Object.entries(defaultStoreValues)) : new Map()
```

Comment:
why map and not an object?

Reply:
Because it was a `Map` in `asynchronous-local-storage`.

In my own context plugin that I'm moving away from now I have it as an object.

Have no strong opinion either way. I do think a `Map` fits in here.

Reply:
based on my prior experience working on a cache, object is going to be faster here, so I would use an opportunity to tweak the implementation.
I wrote the original version a long time ago :)

Reply:
Fixed ✔️